### PR TITLE
Refactor/filtering-methods-ect

### LINKED ISSUE #55: REFACTOR
* [x] ADD Ingredient amounts to bigModal
* [x] DRY up favorite and mealplan methods
* [x] REFACTOR repeated filter methods between classes
* [x] REPLACE event listeners using event delegation
* [x] ADD message on favs page when no favs exist
* [x] FIX image distortion on bigModal
* [x] ADD npm install to readme
* [x] REFACTOR the else/if logic w/i displayCurrentRecipes
* [ ] FIX mealplan and filter button logic
* [x] ADD a mealWasCooked function to User (probably)
* [x] REFACTOR promise handling with helper functions
* [ ] HANDLE errors in the pantry search bar
* [x] UPDATE Readme

### PR DESCRIPTION
**Pull request by ectweitmann**
## What Features are Being Added?
- Nothing new has been added

## What is Being Refactored?
- Refactored the cookbook class filtering methods to be more dynamic, thus removing the need for the User class filter methods.
- Improved search by recipe name functionality to return a broader range of recipe results

## What Do You Need From Your Teammates?
- Review my edits and let me know if anything needs to be removed or improved!

## Any Other Notes/Questions/Problems?
- Not at this time!

### CODE REVIEW

File: src/classes/Cookbook.js [hunk 1]
```diff
@@ -7,8 +7,8 @@ class Cookbook {
     this.keywords = [];
   }
 
-  filterByTag(searchTags) {
-    this.currentRecipes = this.recipes.filter(recipe => {
+  filterByTag(searchTags, recipeList) {
```
Comment (alexmfritz) [inline src/classes/Cookbook.js:10]: THANK YOU for making this so dynamic

File: src/classes/Cookbook.js [hunk 2]
```diff
@@ -34,8 +34,8 @@ class Cookbook {
     }
   }
 
-  clearFilter() {
-    this.currentRecipes = this.recipes;
+  clearFilter(recipeList) {
+    this.currentRecipes = recipeList || this.recipes;
```
Comment (Daniel-OC) [inline src/classes/Cookbook.js:30]: So much cleaner, a BEAUTIFUL fix

File: src/classes/Cookbook.js [hunk 3]
```diff
@@ -30,17 +30,25 @@ class Cookbook {
...
-  filterByIngredient() {
-    if (this.keywords.length) {
-      this.currentRecipes = this.recipes.filter(recipe => {
-        let ingredientNamesArray = recipe.getIngredientNames(this.ingredients).flat();
-        return this.keywords.every(keyword => ingredientNamesArray.includes(keyword));
-      });
-    } else {
-      this.clearFilter();
-    }
+  filterByIngredient(recipeList) {
```
Comment (alexmfritz) [inline src/classes/Cookbook.js:22]: Praise be Ethan

File: src/classes/Cookbook.js [hunk 4]
```diff
@@ -30,17 +30,25 @@ class Cookbook {
...
+    this.currentRecipes = recipeList.filter(recipe => {
+      let ingredientNamesArray = recipe.getIngredientNames(this.ingredients).flat();
+      return this.keywords.every(keyword => ingredientNamesArray.includes(keyword));
```
Comment (Daniel-OC) [inline src/classes/Cookbook.js:25]: Yeah, I think this is better, good work team

**Pull request merged**